### PR TITLE
[#46] importer utils getting a work from the hotfolder and moving it …

### PR DESCRIPTION
**Pull request by pbroman**
…to an import-in-progress folder

### CODE REVIEW

File: kitodo-mediaserver-importer/src/main/java/org/kitodo/mediaserver/importer/util/ImporterUtils.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+        if (hotfolder == null || !hotfolder.isDirectory()) {
+            throw new ImporterException("The hotfolder '" + importerProperties.getHotfolderPath()
+                    + "' is not a directory, please check your configuration");
+        }
+        if (importingFolder == null) {
+            throw new ImporterException("The import-in-progress folder '" + importerProperties.getImportingFolderPath()
+                    + "' is not defined, please check your configuration");
+        }
+        importingFolder.mkdirs();
```
Comment (ponchofiesta) [inline kitodo-mediaserver-importer/src/main/java/org/kitodo/mediaserver/importer/util/ImporterUtils.java]: This is already called on line 59. Is this supposed to be so?

File: kitodo-mediaserver-importer/src/main/java/org/kitodo/mediaserver/importer/util/ImporterUtils.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+        // If there is no xml file directly under the hotfolder, check for a directory
+        Optional<File> optionalDir = FileUtils
+                .listFilesAndDirs(
+                        hotfolder,
+                        FalseFileFilter.FALSE,
+                        TrueFileFilter.TRUE)
+                .stream()
+                .filter(dir -> !dir.equals(hotfolder))
+                .filter(file -> file.isDirectory())
```
Comment (ponchofiesta) [inline kitodo-mediaserver-importer/src/main/java/org/kitodo/mediaserver/importer/util/ImporterUtils.java]: Intellij says: `.filter(File::isDirectory)`

File: kitodo-mediaserver-importer/src/main/java/org/kitodo/mediaserver/importer/control/ImporterFlowControl.java [hunk 3]
```diff
@@ -36,56 +57,58 @@ public void run(String... args) throws Exception {
...
-
-            * Move work folder to the error folder.
-
-            * If there were old data for this work present:
-
-            * Move old work files from the temporary folder back to the original folder.
-
-            * If the error occured writing to the database, rollback (of course).
-         */
+//            * Get a work from the hotfolder and move it to the import-in-progress-folder. Make sure, this set of files is
```
[the remark this answers is not in the document]
Comment (ponchofiesta) [reply kitodo-mediaserver-importer/src/main/java/org/kitodo/mediaserver/importer/control/ImporterFlowControl.java]: Actually it didn't fail... But it should... don't know. So maybe no problem here.

File: kitodo-mediaserver-importer/src/main/java/org/kitodo/mediaserver/importer/config/ImporterConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package org.kitodo.mediaserver.importer.config;
```
[the remark this answers is not in the document]
Comment (pbroman) [reply kitodo-mediaserver-importer/src/main/java/org/kitodo/mediaserver/importer/config/ImporterConfiguration.java]: ok

File: kitodo-mediaserver-importer/src/main/java/org/kitodo/mediaserver/importer/control/ImporterFlowControl.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (pbroman) [reply kitodo-mediaserver-importer/src/main/java/org/kitodo/mediaserver/importer/control/ImporterFlowControl.java]: indentation added

File: kitodo-mediaserver-importer/src/main/java/org/kitodo/mediaserver/importer/util/ImporterUtils.java (hunk 1 above)
Comment (pbroman) [reply kitodo-mediaserver-importer/src/main/java/org/kitodo/mediaserver/importer/util/ImporterUtils.java]: removed line 59

File: kitodo-mediaserver-importer/src/main/java/org/kitodo/mediaserver/importer/util/ImporterUtils.java (hunk 2 above)
Comment (pbroman) [reply kitodo-mediaserver-importer/src/main/java/org/kitodo/mediaserver/importer/util/ImporterUtils.java]: ok

**Pull request merged**